src/server/frontend_wayland/wayland_surface_observer.cpp

```diff
@@ -77,7 +77,11 @@ void mf::WaylandSurfaceObserver::content_resized_to(ms::Surface const*, geom::Si
     run_on_wayland_thread_unless_window_destroyed(
         [content_size](Impl* impl, WindowWlSurfaceRole* window)
         {
-            if (impl->requested_size.value_or(impl->window_size) != content_size)
+            auto const should_resize =
+                (!impl->requested_size && impl->window_size != content_size) ||                                      // No requested resize, current size is different
+                (impl->requested_size && impl->requested_size != content_size && impl->window_size != content_size); // both requested and current sizes are different
```

Comment:
I think this approach is wrong. In essence, it is guessing whether the client has reacted to a configure event by looking at the current size. That will never work for all cases.

As we have sent a configure, we should be tracking the ack_configure (similar to the layer-shell logic you just fixed)

Reply:
(And, yes making the test more complex is tempting. I fell for it when I looked at https://github.com/canonical/mir/issues/3040 a year ago)